File: workspaces/clients/php/workspaces/laravel/tests/Browser/InteractsWithKeyboardTest.php
<?php

namespace tests\Browser;

use Puth\Laravel\Browser;
use Puth\Laravel\Keyboard;
use Tests\Browser\Pages\Playground;
use Tests\PuthTestCase;

class InteractsWithKeyboardTest extends PuthTestCase
{
    public static bool $debug = false;

    function test_keyboard_special_keys()
    {
        $this->browse(function (Browser $browser) {
            $browser->visit(new Playground)
                ->clear('#properties-value input')
                ->keys('#properties-value input', $text = 'äöü', '{ctrl}acvv', '-')
                ->assertInputValue('#properties-value input', $text . $text . '-');
        });
    }
    
    function test_keyboard_with_keyboard()
    {
        $this->browse(function (Browser $browser) {
            $browser->visit(new Playground)
                ->clear('#properties-value input')
                ->keys('#properties-value input', $text = 'äöü')
                ->withKeyboard(function (Keyboard $keyboard) {
                    $keyboard->press('ctrl')
                        ->press('a')
                        ->release('a')
                        ->press('c')
                        ->release('c')
                        ->press('v')
                        ->release('v')
                        ->press('v')
                        ->release('v')
                        ->release('ctrl')
                        ->type('-');
                        
                })
                ->assertInputValue('#properties-value input', $text . $text . '-');
        });
    }
}

File: workspaces/clients/java/client/src/test/java/io/puth/InteractsWithKeyboardTest.java
package io.puth;

import org.junit.jupiter.api.Test;

public class InteractsWithKeyboardTest extends BaseTest {

    @Test
    void test_keyboard_special_keys() {
        String text = "äöü";

        browser.visit(playground())
                .clear("#properties-value input")
                .keys("#properties-value input", new String[]{text, "{ctrl}acvv", "-"})
                .assertInputValue("#properties-value input", text + text + "-");
    }

    @Test
    void test_keyboard_with_keyboard() {
        String text = "äöü";

        browser.visit(playground())
                .clear("#properties-value input")
                .keys("#properties-value input", new String[]{text});
//                .withKeyboard(keyboard -> {
//                    keyboard.press("ctrl")
//                            .press("a").release("a")
//                            .press("c").release("c")
//                            .press("v").release("v")
//                            .press("v").release("v")
//                            .release("ctrl")
//                            .type("-");
//                })
//                .assertInputValue("#properties-value input", text + text + "-");
    }
}
